test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +92,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        }
```

Comment:
Good attempt at updating tests. 👍  Note that a stronger test should also ensure that the file is NOT saved if the command is not mutating. This test only ensures the file is saved when the command is mutating.